src/ILCompiler.ReadyToRun/src/Win32Resources/ResourceData.FileAbstraction.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace ILCompiler.Win32Resources
+{
+    public unsafe partial class ResourceData
+    {
+        /// <summary>
+        /// This is an abstraction around a List of bytes which allows reading and writing the contents
+        /// </summary>
+        private class Mu
```

Comment:
Would be nice if this could be replaced with a standard .NET thing e.g. a `MemoryStream`.

Reply:
Agreed, but the real task is to not use this logic at all, and rebuild it in a simpler fashion. My current intention is to simply check this in, and replace with something more sensible soon.